neurots/morphmath/bifurcation.py

```diff
@@ -19,57 +19,75 @@
 
 import neurots.morphmath.rotation as rt
 from neurots.morphmath.utils import get_random_point
+from neurots.utils import PIA_DIRECTION
 
 
-def random(random_generator=np.random):
+def random(random_generator=np.random, pia_direction=None):  # pylint: disable=unused-argument
     """Get 3-d coordinates of a new random point.
 
     The distance between the produced point and (0,0,0) is given by the value D.
     """
     dir1 = get_random_point(random_generator=random_generator)
     dir2 = get_random_point(random_generator=random_generator)
 
-    return (np.array(dir1), np.array(dir2))
+    return dir1, dir2
 
 
-def symmetric(direction, angles):
+def symmetric(direction, angles, pia_direction=None):
     """Get 3-d coordinates for two new directions at a selected angle."""
-    # phi0 = angles[0] #not used
-    # theta0 = angles[1] #not used
     phi1 = angles[2] / 2.0
     theta1 = angles[3] / 2.0
 
-    dir1 = rt.rotate_vector(direction, [0, 0, 1], phi1)
-    dir1 = rt.rotate_vector(dir1, [1, 0, 0], theta1)
-    dir2 = rt.rotate_vector(direction, [0, 0, 1], -phi1)
-    dir2 = rt.rotate_vector(dir2, [1, 0, 0], -theta1)
+    x_dir = np.array([1, 0, 0])
+    z_dir = np.array([0, 0, 1])
+    if pia_direction is not None:
+        rot = rt.rotation_matrix_from_vectors(PIA_DIRECTION, pia_direction).T
```

Comment:
As far as I understand, the `pia_direction` is only used in this file, so what about computing this rotation matrix earlier and then pass it through all the functions instead of the value of `pia_direction` so we don't compute the rotation matrix again for each bifurcation?

Reply:
yes, that could be better indeed, I'll try after lunch!

Reply:
much cleaner this way!